cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SchemaIndexTest.scala

```diff
@@ -182,13 +254,63 @@ class SchemaIndexTest extends DocumentingTestBase with QueryStatisticsTestSuppor
     )
   }
 
+  @Test def use_index_with_ends_with() {
+    executePreparationQueries {
+      val a = (0 to 100).map { i => "CREATE (:Person)" }.toList
+      val b = (0 to 300).map { i => s"CREATE (:Person {firstname: '$i'})" }.toList
+      a ++ b
+    }
+
+    sampleAllIndexesAndWait()
+
+    profileQuery(
+      title = "Suffix search using `ENDS WITH` (single-property index)",
+      text =
+        "The `ENDS WITH` predicate on `person.firstname` in the following query will use the `Person(firstname)` index, if it exists. " +
+          "All values stored in the `Person(firstname)` index will be searched, and entries ending with `'rk'` will be returned. " +
+          "This means that although the search will not be optimized to the extent of queries using `=`, `IN`, `>`, `<` or `STARTS WITH`, it is still faster than not using an index in the first place. " +
+          "Composite indexes are currently not able to support `ENDS WITH`. ",
+      queryText = "MATCH (person:Person) WHERE person.firstname ENDS WITH 'rk' RETURN person",
+      assertions = {
+        (p) =>
+          assertEquals(1, p.size)
+          assertThat(p.executionPlanDescription().toString, containsString("NodeIndexEndsWithScan"))
+      }
+    )
+  }
+
+  @Test def use_index_with_contains() {
+    executePreparationQueries {
+      val a = (0 to 100).map { i => "CREATE (:Person)" }.toList
+      val b = (0 to 300).map { i => s"CREATE (:Person {firstname: '$i'})" }.toList
+      a ++ b
+    }
+
+    sampleAllIndexesAndWait()
+
+    profileQuery(
+      title = "Substring search using `CONTAINS` (single-property index)",
+      text =
+        "The `CONTAINS` predicate on `person.firstname` in the following query will use the `Person(firstname)` index, if it exists. " +
+          "All values stored in the `Person(firstname)` index will be searched, and entries containing `'r'` will be returned. " +
+          "This means that although the search will not be optimized to the extent of queries using `=`, `IN`, `>`, `<` or `STARTS WITH`, it is still faster than not using an index in the first place. " +
+          "Composite indexes are currently not able to support `CONTAINS`. ",
+      queryText = "MATCH (person:Person) WHERE person.firstname CONTAINS 'r' RETURN person",
+      assertions = {
+        (p) =>
+          assertEquals(2, p.size)
+          assertThat(p.executionPlanDescription().toString, containsString("NodeIndexContainsScan"))
+      }
+    )
+  }
+
   @Test def use_index_with_exists_property() {
     // Need to make index preferable in terms of cost
     executePreparationQueries((0 to 250).map { i =>
       "CREATE (:Person)"
     }.toList)
     profileQuery(
-      title = "Use index when checking for the existence of a property",
+      title = "Existence check using `exists` (single-property index)",
```

Comment:
I thought the old title read much better.

Reply:
I think the old title took too long to actually explain what was happening. Here, one sees from the outset what one can do - it's implicit that the "index is being used"

Reply:
Then my only remaining complaint is the `(single-property index)`. But, as you said before, the Docs team could perhaps take a decision on that.